src/OceanSeaIceModels/ocean_sea_ice_model.jl

```diff
@@ -59,17 +61,33 @@ prettytime(model::OSIM)             = prettytime(model.clock.time)
 iteration(model::OSIM)              = model.clock.iteration
 timestepper(::OSIM)                 = nothing
 default_included_properties(::OSIM) = tuple()
-prognostic_fields(cm::OSIM)         = nothing
+prognostic_fields(::OSIM)           = nothing
 fields(::OSIM)                      = NamedTuple()
 default_clock(TT)                   = Oceananigans.TimeSteppers.Clock{TT}(0, 0, 1)
+time(model::OSIM)                   = model.clock.time
+checkpointer_address(::OSIM)        = "HydrostaticFreeSurfaceModel"
 
-function reset!(model::OSIM)
-    reset!(model.ocean)
-    return nothing
-end
+reset!(model::OSIM) = reset!(model.ocean)
+
+initialize!(model::OSIM) = initialize!(model.ocean)
+
+initialize_jld2_file!(filepath, init, jld2_kw, including, outputs, model::OSIM) =
+    initialize_jld2_file!(filepath, init, jld2_kw, including, outputs, model.ocean.model)
+
+write_output!(c::Checkpointer, model::OSIM) = write_output!(c, model.ocean.model)
+
+function set!(sim::OSIMSIMPA, pickup::Union{Bool, Integer, String})
+    checkpoint_file_path = checkpoint_path(pickup, sim.output_writers)
+
+    set!(sim.model.ocean.model, checkpoint_file_path)
+
+    sim.model.clock.iteration = sim.model.ocean.model.clock.iteration
+    sim.model.clock.time = sim.model.ocean.model.clock.time
+
+    # Setting the atmosphere time to the ocean time
+    sim.model.atmosphere.clock.iteration = sim.model.ocean.model.clock.iteration
+    sim.model.atmosphere.clock.time = sim.model.ocean.model.clock.time
```

Comment:
Rather than extending set! For every atmos model, it might be better to write a function for setting the clock and then extend that appropriately

Reply:
Same comment for ocean and sea ice

Reply:
As in:

set_clock!(atmos, time, iter)

Reply:
Good point regarding defining a `set_clock!` method.

Just slightly confused: this doesn't alleviate the need to extend `set!`, right?

Reply:
I don't think so @navidcy - currently the set! function in Oceananigans does `set!(sim.model, checkpoint_file_path)` when we want it to do `set!(sim.model.ocean.model, checkpoint_file_path)` because of the coupled nature of the model.

Reply:
so you are saying that we can do this _without_ extending `set!`?

Reply:
Sounds like you need to define `set!(::OceanSeaIceMode, filepath)`.

Reply:
No, I think we need to extend `set!` and create the function `set_clock!` as @glwagner suggests. See my latest push which has added that function.

Reply:
One problem with `set!(::OSIMSIM, filepath)` is that, when we eventually support checkpointing diagnostics  / callbacks / output writer states, that support will have to be copy/pasted here. So it'd be better to use `set!(::OceanSeaIceModel, filepath)`.

Reply:
OK. But given then the output writers/checkpointers are included in the outer-outer simulations, then if we define `set!(::OSIM, filepath)` that won't have access to the checkpointer. Or? Am I missing something?

Reply:
Otherwise, I like the suggestion!

Reply:
```
set!(::OceanSeaIceModel, filepath)
```
done via [4110aca](https://github.com/CliMA/ClimaOcean.jl/pull/401/commits/4110aca3ce4c0f6adf767d9fdaebd0be825504ef)